src/ripple/rpc/impl/LookupLedger.cpp

```diff
@@ -47,35 +64,58 @@ Status ledgerFromRequest (
 
     if (!hashValue.empty())
     {
+        if (! hashValue.isString ())
+            return {rpcINVALID_PARAMS, "ledgerHashNotString"};
+
         uint256 ledgerHash;
-        if (hashValue.isString() && ledgerHash.SetHex (hashValue.asString ()))
-            ledger = netOps.getLedgerByHash (ledgerHash);
-        else
+        if (! ledgerHash.SetHex (hashValue.asString ()))
             return {rpcINVALID_PARAMS, "ledgerHashMalformed"};
+           
+        if ((ledger = netOps.getLedgerByHash (ledgerHash)) == nullptr)
+        {
+            auto cl = netOps.getCurrentLedger ();
+            if (cl == nullptr || ! synced (netOps, cl))
+                return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+
+            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
+        }
     }
     else if (indexValue.isNumeric())
     {
-        ledger = netOps.getLedgerBySeq (indexValue.asInt());
+        if ((ledger = netOps.getLedgerBySeq (indexValue.asInt ())) == nullptr)
+        {
+            auto cl = netOps.getCurrentLedger ();
+            if (cl == nullptr || ! synced (netOps, cl))
+                return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+
+            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
+        }
     }
     else
     {
-        auto index = indexValue.asString();
-        auto isCurrent = index.empty() || index == "current";
+        auto const index = indexValue.asString ();
```

Comment:
And for the opposing viewpoint... :-)  I think `const` adds value.  If I want to understand the code, it helps if I know which variables are changing and which or not, and it is quicker to read the `const` on the variable declaration than search the remaining code for statements that might modify the variable.

This is Scott Meyers Item 3 from Effective C++:  Use `const` whenever possible.  I'm the first to admit that I won't blindly follow everything Scott has, and I do have disagreements with some of his advice.  But not this one.

In C++14 (VS willing), the advice gets even stronger:  Use `constexpr` whenever possible.  This can have significant run-time performance benefits.  Clarification:  Neither of these cases is one where we could drop in `constexpr`.  But there are lots of other places in rippled where we can and should.